src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -0,0 +1,793 @@
+            "cell", lfric_psyir.CellPositionDataSymbol)
+        self._arglist.append(symbol)
+
+    def mesh_height(self, var_accesses=None):
+        '''Create an LFRic mesh height object and add it to the symbol table
+        and argument list.
+
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        '''
+        symbol = self._create_symbol(
+            "nlayers", lfric_psyir.MeshHeightDataSymbol)
+        self._arglist.append(symbol)
+
+    def mesh_ncell2d(self, var_accesses=None):
+        '''Not implemented.
+
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: as this method is not implemented.
+
+        '''
+        raise NotImplementedError("mesh_ncell2d not implemented")
+
+    def cell_map(self, var_accesses=None):
+        '''Not implemented.
+
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: as this method is not implemented.
+
+        '''
+        raise NotImplementedError("cell_map not implemented")
+
+    def field_vector(self, argvect, var_accesses=None):
+        '''Create LFRic field vector arguments and add them to the symbol
+        table and argument list. Also declare the associated "undf"
+        symbol if it has not already been declared so that it can be
+        used to dimension the field vector arguments.
+
+        :param argvect: the field vector to add.
+        :type argvect: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: if the datatype of the vector \
+            field is not supported.
+
+        '''
+        # self._create_symbol(...) will create and add a undf symbol
+        # to the symbol table if one does not already exist or return
+        # the existing one if one does.
+        fs_name = argvect.function_space.orig_name
+        undf_symbol = self._create_symbol(
+            "undf_{0}".format(fs_name),
+            lfric_psyir.NumberOfUniqueDofsDataSymbol,
+            extra_args=[fs_name])
+
+        interface = ArgumentInterface(INTENT_MAPPING[argvect.intent])
+        try:
+            field_class = self.vector_field_mapping[
+                argvect.intrinsic_type]
+        except KeyError:
+            raise NotImplementedError(
```

Comment:
pylint says "consider using raise-from" here. I'm not sure whether it would be useful?

Reply:
OK, I've changed to using six.raise_from and modified two other cases in this file where I added a pylint ignore directive. Could you let me know what version of pylint you are using. I just pulled the latest ubuntu version in WSL2 using apt install pylint ... which says it is 2.4.4 but this does not pick up the error or recognise the pylint ignore directive.